### PR TITLE
CLN: Use C-API for datetime.date

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jbrockmendel** [comment]
LGTM

**Comment by TomAugspurger** [comment]
For my own understanding, why do we prefer to use the C API? Is this likely to improve performance?

**Comment by MomIsBestFriend** [comment]
>  Is this likely to improve performance?

Correct! 

---

This is because when we use ```import``` and not ```cimport``` we are going through the python space, which is slower than pure C.

I have made POC so you could see to diffs:

date_vs_Cdate

Both are doing the exact same thing, only that the C-API ```datetime.date``` is much faster.

File: pandas/_libs/tslibs/strptime.pyx [hunk 1]
```diff
@@ -4,7 +4,6 @@ import time
 import locale
 import calendar
 import re
-import datetime
```
Comment (WillAyd) [inline pandas/_libs/tslibs/strptime.pyx:7]: I think can just cimport cpython.datetime as datetime to minimize changes no? Didn't we move away from the import pattern here already in #32459

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/_libs/tslibs/strptime.pyx (hunk 1 above)
Comment (MomIsBestFriend) [reply pandas/_libs/tslibs/strptime.pyx:7]: Good catch!

Somehow slipped away from me

**Review by WillAyd — APPROVED** [review]
lgtm

**Comment by mroeschke** [comment]
Thanks @MomIsBestFriend

**Pull request merged**